### PR TITLE
Adds support for Question Palette

### LINKED ISSUE #44: Add support for question palette

### PR DESCRIPTION
**Pull request by dalmia**
Fixes #44 

Requires: https://github.com/avantifellows/quiz-backend/pull/37

## Summary
- adds hamburger for toggling question palette
- adds components for success/error/neutral icons shown inside the palette
- shows different icons for success/error/neutral based on whether the quiz has ended
- creates a new variable called `questionStates` which stores the state of each graded question - either success/error/neutral.
- adds a palette item component which represents each question shown in the palette - shows the appropriate success/error/neutral state for each question based on the question's state as specified in the `questionStates` variable mentioned above + highlights the current question
- added click-away to automatically close palette when someone clicks away from the hamburger icon
- add utility function to evaluate whether the answer to a given question is correct. 
- keeps track of whether a given question has been visited or not.

## Test Plan

- [x] Test Responsiveness
  - [ ] Laptop (1200px)
  - [ ] Tablet (760px)
  - [ ] Phone (320px)
- [x] Cross-Browser Testing
  - [ ] Chrome
  - [ ] Firefox
- [ ] ~Local Language Support~
- [x] Hygiene and Housekeeping
  - [ ] Self-review
  - [ ] Comments have been added appropriately
  - [ ] ~Check for bundle size [here](https://bundlephobia.com/) if adding a package~
  - [ ] Added relevant details like Labels/Projects/Milestones etc.
  - [ ] ~If adding or removing any environment variable, update `docs/ENV.md`, `.env.example` and the Github workflows.~
- [ ] Testing
  - [x] Wrote tests
  - [x] Tested locally
  - [x] Tested on staging
  - [ ] Tested on an actual physical phone
  - [ ] Tested on production
- [x] Lighthouse Checks
  - [ ] ~Images have `alt` attributes~
  - [ ] ~Any `` tags have `width` and `height` specified~
  - [ ] ~Any `target="_blank"` links have `rel="noopener"`~
  - [ ] Only SVGs are used as images. If PNGs are used, their size has been optimised.
  - [ ] ~Any SVG buttons without text have their `aria-label` attributes set~

### CODE REVIEW

File: src/components/Questions/Body.vue [hunk 1]
```diff
@@ -376,6 +407,15 @@ export default defineComponent({
...
     );
+    const legendSuccessText = computed(() =>
+      props.hasQuizEnded ? "Correct" : "Answered"
+    );
+    const legendErrorText = computed(() =>
+      props.hasQuizEnded ? "Wrong" : "Not Answered"
+    );
+    const legendNeutralText = computed(() =>
+      props.hasQuizEnded ? "Skipped" : "Not Visited"
+    );
```
Comment (dalmia) [inline src/components/Questions/Body.vue:418]: delete

**Comment by deepansh96** [comment]
- once I clear an existing answer, the clear button should get disabled

File: src/components/Questions/Header.vue [hunk 2]
```diff
@@ -21,30 +30,74 @@ export default defineComponent({
...
+    isPaletteVisible: {
+      type: Boolean,
+      default: false,
+    },
   },
   setup(props, context) {
     const state = reactive({
       endTestButtonClass:
         "bg-emerald-500 hover:bg-emerald-600 ring-emerald-500 p-2 px-4 bp-500:p-4 bp-500:px-6 rounded-lg sm:rounded-2xl shadow-xl disabled:opacity-50 disabled:pointer-events-none",
+      localIsPaletteVisible: props.isPaletteVisible,
```
Comment (deepansh96) [inline src/components/Questions/Header.vue:43]: why do you need a local version?

Comment (deepansh96) [reply src/components/Questions/Header.vue:43]: because we cannot mutate props?
v-model laga de?

File: src/services/Functional/Utilities.ts [hunk 3]
```diff
@@ -63,3 +66,51 @@ export function resetConfetti() {
     cancelAnimationFrame(animationFrameRequest);
   }
 }
+
+/**
+ * Given a question and its corresponding answer, this method checks if the answer is correct.
+ * If the question is ungraded, the method returns that the evaluation is invalid.
+ * For graded questions, it returns whether the question has been answered and if it has been,
+ * is the answer correct.
```
Comment (deepansh96) [inline src/services/Functional/Utilities.ts:74]:
```suggestion
 * it also returns whether the answer is correct.
```

**Comment by dalmia** [comment]
> once I clear an existing answer, the clear button should get disabled

Made an issue for it: https://github.com/avantifellows/quiz-frontend/issues/53

**Pull request merged**